### DISCUSSION TITLE
AttributeError: 'CRNN' object has no attribute 'load_state_dict'

### DISCUSSION THREAD

**Discussion by anhalu** [mindee/doctr#1338] [Q&A] [answered]
### Bug description

I trained a crnn_vgg16_bn recog model, But now, I can't load it to ocr_predictor, I use pytorch and install with cmd "pip install "python-doctr[torch]"  

### Code snippet to reproduce the bug

import torch 
import doctr
from doctr.models import ocr_predictor, db_resnet50, crnn_vgg16_bn

model = ocr_predictor(det_arch = 'db_resnet50', reco_arch='crnn_vgg16_bn', pretrained=True) 

model.reco_predictor.model.load_state_dict(torch.load('/hkaggle/input/recog-model/crnn_vgg16_bn.pt')) 

### Error traceback

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[17], line 14
      4 # reco_model = crnn_vgg16_bn(pretrained=False, pretrained_backbone=False)
      5 # reco_params = torch.load('/kaggle/input/recog-model/crnn_vgg16_bn.pt', map_location="cpu")
      6 # # reco_model.load_state_dict(reco_params)
      7 # reco_model.model.load_state_dict(reco_params)
      8 # predictor = ocr_predictor(det_arch="linknet_resnet18", reco_arch=reco_model, pretrained=True)
     12 model = ocr_predictor(det_arch = 'db_resnet50', reco_arch='crnn_vgg16_bn', pretrained=True) 
---> 14 model.reco_predictor.model.load_state_dict(torch.load('/hkaggle/input/recog-model/crnn_vgg16_bn.pt')) 

AttributeError: 'CRNN' object has no attribute 'load_state_dict'
add Codeadd Markdown

### Environment

DocTR version: v0.7.0
TensorFlow version: 2.12.0
PyTorch version: 2.0.0 (torchvision 0.15.1)
OpenCV version: 4.8.0
OS: Ubuntu 22.04.2 LTS
Python version: 3.10.12
Is CUDA available (TensorFlow): Yes
Is CUDA available (PyTorch): Yes
CUDA runtime version: 11.8.89
GPU models and configuration: GPU 0: Tesla P100-PCIE-16GB
Nvidia driver version: 470.161.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.0

### Deep Learning backend

is_tf_available: True
is_torch_available: True

**Comment by anhalu** [comment]
@felixdittrich92  can you help me, plz

**Comment by felixdittrich92** [comment, marked as answer]
Hi @anhalu :wave,

You have installed both TensorFlow and PyTorch on your machine this is only recommended for testing and development.
docTR firstly tries to import tensorflow that's also the reason of your error message.

docs: https://mindee.github.io/doctr/using_doctr/custom_models_training.html#loading-your-custom-trained-model

```
# Load custom recognition model
reco_model = crnn_vgg16_bn(pretrained=False, pretrained_backbone=False)
reco_params = torch.load('<path_to_pt>', map_location="cpu")
reco_model.load_state_dict(reco_params)
predictor = ocr_predictor(det_arch="linknet_resnet18", reco_arch=reco_model, pretrained=True)
```

How to handle it:

run with pytorch: `USE_TORCH=1 python path/to_your_script.py`
run with tensorflow: `USE_TF=1 python path/to_your_script.py`

**Reply by anhalu** [reply]
@felixdittrich92  Thanks u so much for reply me. But I have a question, when i train model, it raise an error "RuntimeError: Expected tensor to have size at least 101 at dimension 1, but got size 32 for argument #2 'targets' (while checking arguments for ctc_loss_gpu)" , Only when I change the parameter of max_length in "doctr/doctr/models/recognition/crnn/pytorch.py", i can train the model. Is there any way to train the model without editing the code in the library?

**Reply by anhalu** [reply]
I don't see any argument about the max_length in train_pytorch, Am I doing something wrong ?

**Reply by felixdittrich92** [reply]
Hey @anhalu what's the longest word in your custom dataset ? 
The most models have by default a max length of 30-32.
(Keep in mind docTR provides only word level detection/recognition no line level) :)
